xchange-coinmate/src/main/java/org/knowm/xchange/coinmate/service/CoinmateTradeService.java

```diff
@@ -67,8 +68,22 @@ public OpenOrders getOpenOrders(OpenOrdersParams params) throws IOException {
     ArrayList<Order> result = new ArrayList<>(orderQueryParams.length);
     for (OrderQueryParams orderQueryParam : orderQueryParams) {
       CoinmateOrders response = this.getCoinmateOrderById(orderQueryParam.getOrderId());
-      List<Order> orders = CoinmateAdapters.adaptOrders(response);
-      result.addAll(orders);
+      Order order = CoinmateAdapters.adaptOrder(response.getData());
+      if (order.getStatus() == Order.OrderStatus.STOPPED) {
+        // fetch generated market order and get its average price and amount
+        CoinmateOrders marketOrderRaw = this.getCoinmateOrderById( response.getData().getStopLossOrderId());
+        if (marketOrderRaw != null) {
+          order.setAveragePrice(marketOrderRaw.getData().getAvgPrice());
```

Comment:
it's unfortunate that some added setter to `Order` which should be immutable

this results in adaptation being done outside of the adapter

I think it would be better to make the adapter method look like this:

```
public static Order adaptOrder(CoinmateOrder coinmateOrder, Function<CoinmateOrder, String> orderByIdFetcher) {
```

move the logic below to it

and call the adapter like this

```
Order order = CoinmateAdapters.adaptOrder(response.getData(), orderId -> this.getCoinmateOrderById(orderId));
```

Reply:
@walec51 I have modified it with it with Function. One downside is that `Fuction` cannot throw so I added a catch block and return `null` instead. Is there a better solution?

Reply:
Agreed. it should be immutable.